Compiling on GCC12 failed

GCC version: gcc 12.1.1
x16-emulator version: 40

Build log:
```

cc -std=c99 -O3 -Wall -Werror -g -I/usr/include/SDL2 -D_REENTRANT -Isrc/extern/include -Isrc/extern/src -c src/cpu/fake6502.c -o build/cpu/fake6502.o
cc -std=c99 -O3 -Wall -Werror -g -I/usr/include/SDL2 -D_REENTRANT -Isrc/extern/include -Isrc/extern/src -c src/memory.c -o build/memory.o
cc -std=c99 -O3 -Wall -Werror -g -I/usr/include/SDL2 -D_REENTRANT -Isrc/extern/include -Isrc/extern/src -c src/disasm.c -o build/disasm.o
cc -std=c99 -O3 -Wall -Werror -g -I/usr/include/SDL2 -D_REENTRANT -Isrc/extern/include -Isrc/extern/src -c src/video.c -o build/video.o
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 64 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 80 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 96 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 112 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 128 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 144 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 160 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 176 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 192 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
In function 'expand_4bpp_data',
    inlined from 'render_sprite_line' at src/video.c:512:4,
    inlined from 'render_line' at src/video.c:849:3:
src/video.c:472:22: error: writing 16 bytes into a region of size 0 [-Werror=stringop-overflow=]
  472 |                 *dst = (*src) & 0xf;
      |                 ~~~~~^~~~~~~~~~~~~~
src/video.c: In function 'render_line':
src/video.c:509:25: note: at offset 208 into destination object 'unpacked_sprite_line' of size 64
  509 |                 uint8_t unpacked_sprite_line[64];
      |                         ^~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:74: build/video.o] Error 1
```

--- a/src/video.c
+++ b/src/video.c
@@ -18,6 +18,7 @@
 #include "sdcard.h"
 
 #include <limits.h>
+#include <stdint.h>
 
 #ifdef __EMSCRIPTEN__
 #include "emscripten.h"
@@ -507,12 +508,13 @@ render_sprite_line(const uint16_t y)
 		const uint8_t *bitmap_data = video_ram + props->sprite_address + (eff_sy << (props->sprite_width_log2 - (1 - props->color_mode)));
 
 		uint8_t unpacked_sprite_line[64];
+		const uint16_t width = (props->sprite_width<64? props->sprite_width : 64);
 		if (props->color_mode == 0) {
 			// 4bpp
-			expand_4bpp_data(unpacked_sprite_line, bitmap_data, props->sprite_width);
+			expand_4bpp_data(unpacked_sprite_line, bitmap_data, width);
 		} else {
 			// 8bpp
-			memcpy(unpacked_sprite_line, bitmap_data, props->sprite_width);
+			memcpy(unpacked_sprite_line, bitmap_data, width);
 		}
 
 		for (uint16_t sx = 0; sx < props->sprite_width; ++sx) {

